stonesoup/metricgenerator/manager.py

```diff
@@ -78,17 +78,16 @@ def generate_metrics(self):
         if self.associator is not None and self.association_set is None:
             self.associate_tracks()
 
-        metrics = set()
+        metrics = {}
         for generator in self.generators:
-            metric = generator.compute_metric(self)
-            # Metrics can be lists or not, there's probably a neater way to do
-            # this
-            if isinstance(metric, list):
-                metrics.update(metric)
-            else:
-                metrics.add(metric)
-
-        return set(metrics)
+            metric_list = generator.compute_metric(self)
+            # If not already a list, force it to be one below
+            if not isinstance(metric_list, list):
+                metric_list = [metric_list]
+            for metric in metric_list:
+                key = metric.title.replace(' at times', '')  # Shorten key a little
```

Comment:
We'd end-up with a repeated key for SIAPs here. Eg. "SIAP Completeness" (which is an average completeness value over all timestamps) and "SIAP Completeness at times" (which is a list of completeness values, with a value for every timestamp).

Reply:
Ah yes I realised this late yesterday - I'll keep the ' at times'.